### ISSUE TITLE
Flaky Tests Invalid server state: initialized

### ISSUE THREAD

**Issue by dplewis** [parse-community/parse-server#9149]
### New Issue Checklist

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Before posting search [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

The test suite in this repo and in the JS SDK both have flaky tests. I believe this is due to the asyncronous nature of starting the server instance.

```
await ParseServer.startApp(newConfiguration);
```
I think the issue is after the promise resolves the server is still initializing. https://github.com/parse-community/parse-server/pull/8232

```
Parse.Query testing dontSelect query without conditions
  - Unhandled promise rejection: ParseError: 111 Invalid server state: initialized,ParseError: 111 Invalid server state: initialized,ParseError: 111 Invalid server state: initialized,ParseError: 111 Invalid server state: initialized,ParseError: 111 Invalid server state: initialized  
```

### Steps to reproduce

I can't reproduce this locally

### Actual Outcome

Tests fail

### Expected Outcome

Test Pass

**Comment by mtrezza** [comment]
So `initialized` is the state after calling the `ParseServer` constructor. As soon as you call `start()` on the instance, the state changes to `starting`. From what you describe it looks like `start()` has not been called at all, because `starting` is set synchronously after calling `start()`. So it may be related to the `reconfigureServer` logic; that promise may return early, rather than the server init itself. Which would be good, because that means it's only a testing issue, not a server issue.

**Comment by mtrezza** [comment]
Closing via https://github.com/parse-community/Parse-SDK-JS/pull/2183

**Comment by mtrezza** [comment]
Re-opened, I think we first need to upgrade the Parse JS SDK in Parse Server to see whether the Parse Server CI shows less flakiness.

**Comment by dplewis** [comment]
I agree, I still don’t know why this is happening. I would like to think I solved it on the JS SDK

![IMG_0502](https://github.com/parse-community/parse-server/assets/9830365/9865a8c7-e52a-460a-a9c4-32a2397c69b5)

**Comment by mtrezza** [comment]
Issue still occurs in recent PRs; I have never seen this before; is this a recently introduced issue?

**Comment by dplewis** [comment]
@mtrezza Looks like the server shutsdown and restarts randomly after a few hundred tests. If we turn off the randomized tests we should find a pattern.

**Comment by mtrezza** [comment]
Okay, if you want to open a PR to temporarily disable random tests I'll go ahead and merge

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.2.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.2.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.2.0](https://github.com/parse-community/parse-server/releases/tag/7.2.0)
